File: KtvAssistant_ios/SinaKTVAide/Model/DataStruct/KTVHongBaoInfo.m
//
//  KTVHongBaoInfo.m
//  SinaKTVAide
//
//  Created by 顾利平 on 15/5/21.
//  Copyright (c) 2015年 tiange. All rights reserved.
//

#import "KTVHongBaoInfo.h"
#import "PCommonUtil.h"

@implementation KTVHongBaoInfo

-(instancetype)init
{
    self = [super init];
    if(self)
    {
        self.hbid = 0;
        self.hbnclass = -1;
        self.hbclass = @"";
        self.hbmission = @"";
        self.hbstatus = hongbao_todo;
        self.hbtime = @"";
    }
    
    return self;
}

+(instancetype)initWithDic:(NSDictionary*) data
{
    KTVHongBaoInfo* info = [[KTVHongBaoInfo alloc] init];
    if(info)
    {
        info.hbid = [data objectForKey:@"ID"];
        info.hbnclass = [[data objectForKey:@"RewardType"] integerValue];
        switch(info.hbnclass)
        {
            case 0:
                info.hbclass = @"新人奖励";
                break;
            case 1:
                info.hbclass = @"每日奖励";
                break;
            case 2:
                info.hbclass = @"个人红包";
                break;
        }
        info.hbmission = [PCommonUtil decodeBase64:[data objectForKey:@"DescrInfo"]];
        info.hbstatus = (eHongBaoStatus)[[data objectForKey:@"Status"] integerValue];
        
        id obj = [data objectForKey:@"AddTime"];
        if(obj)
            info.hbtime = [PCommonUtil decodeBase64:obj];
        
        info.gold = [[data objectForKey:@"RewardGold"] integerValue];
    }
    
    return info;
}

@end

File: KtvAssistant_android/src/assistant/entity/KtvHongBaoInfo.java
package assistant.entity;

import org.json.JSONException;
import org.json.JSONObject;

import assistant.task.PCommonUtil;

public class KtvHongBaoInfo {
	//0-新人奖励，1-每日奖励，2-个人红包
	public int hbnclass = -1;
	public String hbclass = "";//红包分类
	public String hbmission = "";//红包说明
	//0-未完成，1-完成，待领取，2-已领取
	public int hbstatus = 0;//红包状态
	public String hbid = "";//红包ID
	public String hbtime = "";//红包时间
	public int gold = 0;//红包金额
	
	public KtvHongBaoInfo(JSONObject jsonObject){
		super();
		
		if(jsonObject != null)
		{
			try {
				hbid = jsonObject.getString("ID");
				hbnclass = jsonObject.getInt("RewardType");
				switch(hbnclass)
				{
					case 0:
					{
						hbclass = "新人红包";
						break;
					}
					case 1:
					{
						hbclass = "每日奖励";
						break;
					}
					case 2:
					{
						hbclass = "个人红包";
						break;
					}
				}
				
				String str = jsonObject.getString("DescrInfo");
				if(str != null)
					hbmission = PCommonUtil.decodeBase64(str);
				hbstatus = jsonObject.getInt("Status");
				str = jsonObject.getString("AddTime");
				if(str != null)
					hbtime = PCommonUtil.decodeBase64(str);
				gold = jsonObject.getInt("RewardGold");
				
			} catch (JSONException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}	
		}
	}
}
